examples/miscellaneous/plot_outlier_detection_bench.py

```diff
@@ -0,0 +1,186 @@
+"""
+==========================================
+Evaluation of outlier detection estimators
+==========================================
+
+This example benchmarks outlier detection algorithms, Local Outlier
+Factor (LOF) and Isolation Forest (IForest), using ROC curves on
+classical anomaly detection datasets. The algorithm performance
+is assessed in an outlier detection context:
+
+1. The algorithms are trained on the whole dataset which is assumed to
+contain outliers.
+
+2. The ROC curve is computed on the same dataset using the knowledge
+of the labels.
+
+There is no particular reason to choose algorithms LOF and IForest. The goal
+is to show that different algorithm performs well on different datasets.
+"""
+
+# Author: Pharuj Rajborirug <pharuj.ra@kmitl.ac.th>
+# License: BSD 3 clause
+
+import numpy as np
+
+print(__doc__)
+
+rng = np.random.RandomState(42)
+
+# %%
+# Generate datasets
+# -----------------
+#
+# The example uses real world datasets available in `scikit-learn.datasets`.
+datasets = [
+    "http",
+    "smtp",
+    "SA",
+    "SF",
+    "forestcover",
+    "glass",
+    "wdbc",
+    "cardiotocography",
+]
+
+# %%
+# Generate outlier detection models
+# ---------------------------------
+
+from sklearn.neighbors import LocalOutlierFactor
+from sklearn.ensemble import IsolationForest
+
+models = [
+    ("LOF", LocalOutlierFactor(n_neighbors=20, contamination="auto")),
+    ("IForest", IsolationForest(random_state=rng, contamination="auto")),
+]
+
+# %%
+# Preprocess data and compute ROC curves
+# --------------------------------------
+#
+# During the data preprocessing, we reduce the sample size of some datasets to
+# speed up computation.
```

Comment:
Following my previous comment, I think this cell would be clearer if it was just doing the plot.